geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -462,6 +486,9 @@ object FilterHelper {
           inverted
         } else if (inverted.disjoint) {
           FilterValues(Seq(Bounds.everything[T])) // equivalent to not null
+        } else if (!inverted.precise) {
+          FilterHelperLogger.log.warn(s"Falling back to full table scan for inverted query: '${filterToString(f)}'")
```

Comment:
For my knowledge, what's an example of a filter which would hit here?

Reply:
basically anything with a function, e.g. `strToUpperCase(name) = 'BILL'`. `precise` differentiates between that case, where we have to scan the whole table but aren't actually returning everything, with `is not null`, where we have to scan the whole table. If the filter isn't precise, we can't really invert it since we don't know which ranges we'd have to scan. let me know if that doesn't make sense, i wrote this a while ago...

Reply:
Seems reasonable.